opensearch/src/main/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactory.java

```diff
@@ -193,8 +189,11 @@ private ExprValue parse(
       return ExprNullValue.of();
     }
 
-    ExprType type = fieldType.get();
-    if (type.equals(OpenSearchDataType.of(OpenSearchDataType.MappingType.Nested))
+    final ExprType type = fieldType.get();
+
+    if (type.equals(OpenSearchDataType.of(OpenSearchDataType.MappingType.GeoPoint))) {
+      return parseGeoPoint(content, supportArrays);
```

Comment:
have u evaluate add this logic in parseArray?
the reason i am asking is parseArray has ` return parseInnerArrayValue(content.array().next(), prefix, type, supportArrays);`

and we can add GeoPoint process in 
```
  private ExprValue parseInnerArrayValue(
      Content content, String prefix, ExprType type, boolean supportArrays) {
    if (type instanceof OpenSearchIpType
        || type instanceof OpenSearchBinaryType
        || type instanceof OpenSearchDateType
        || type instanceof OpenSearchGeoPointType) {
      return parse(content, prefix, Optional.of(type), supportArrays);
    } else if (content.isString()) {
      return parse(content, prefix, Optional.of(OpenSearchDataType.of(STRING)), supportArrays);
    } else if (content.isLong()) {
      return parse(content, prefix, Optional.of(OpenSearchDataType.of(LONG)), supportArrays);
    } else if (content.isFloat()) {
      return parse(content, prefix, Optional.of(OpenSearchDataType.of(FLOAT)), supportArrays);
    } else if (content.isDouble()) {
      return parse(content, prefix, Optional.of(OpenSearchDataType.of(DOUBLE)), supportArrays);
    } else if (content.isNumber()) {
      return parse(content, prefix, Optional.of(OpenSearchDataType.of(INTEGER)), supportArrays);
    } else if (content.isBoolean()) {
      return parse(content, prefix, Optional.of(OpenSearchDataType.of(BOOLEAN)), supportArrays);
    } else {
      return parse(content, prefix, Optional.of(STRUCT), supportArrays);
    }
  }
```

Reply:
Yes, I did. The geo point value can be represented as an array (`[lat, lon]`) or as multiple values. It can be difficult to process this in `parseArray`.